flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/runtime/typeutils/LinkedListSerializer.java

```diff
@@ -110,17 +122,42 @@ public int getLength() {
     @Override
     public void serialize(LinkedList<T> list, DataOutputView target) throws IOException {
         target.writeInt(list.size());
+        if (hasNullMask) {
+            MaskUtils.writeMask(getNullMask(list), target);
+        }
         for (T element : list) {
-            elementSerializer.serialize(element, target);
+            if (element != null) {
+                elementSerializer.serialize(element, target);
+            }
         }
     }
 
+    private boolean[] getNullMask(LinkedList<T> list) {
+        if (reuseMask == null || reuseMask.length != list.size()) {
```

Comment:
`reuseMask.length < list.size()`
We can introduce a `void writeMask(boolean[] mask, int length, DataOutputView target)` to `MaskUtils`.